libopenage/gui/guisys/private/platforms/context_extraction_x11.cpp

```diff
@@ -35,4 +35,23 @@ std::tuple<QVariant, WId> extract_native_context(SDL_Window *window) {
 	);
 }
 
+std::tuple<QVariant, std::function<void()>> extract_native_context_and_switchback_func(SDL_Window *window) {
+	assert(window);
+
+	GLXContext current_context;
+	SDL_SysWMinfo wm_info;
+	SDL_VERSION(&wm_info.version);
+
+	if (SDL_GetWindowWMInfo(window, &wm_info)) {
+		assert(wm_info.info.x11.display);
+
+		current_context = glXGetCurrentContext();
+		assert(current_context);
+	}
```

Comment:
if this if is skipped, we should abort, right? (same above in the old function i think)